server/image_browse_resource.py

```diff
@@ -0,0 +1,56 @@
+from girder.api import access
+from girder.api.v1.item import Item as ItemResource
+from girder.api.describe import autoDescribeRoute, Description
+from girder.constants import AccessType
+from girder.exceptions import RestException
+from girder.models.folder import Folder
+
+
+class ImageBrowseResource(ItemResource):
+    """Extends the "item" resource to iterate through images im a folder."""
+
+    def __init__(self, apiRoot):
+        # Don't call the parent (Item) constructor, to avoid redefining routes,
+        # but do call the grandparent (Resource) constructor
+        super(ItemResource, self).__init__()
+
+        self.resourceName = 'item'
+        apiRoot.item.route('GET', (':id', 'next_image'), self.getNextImage)
+        apiRoot.item.route('GET', (':id', 'previous_image'), self.getPreviousImage)
+
+    def getAdjacentImages(self, currentImage):
+        folderModel = Folder()
+        folder = folderModel.load(
+            currentImage['folderId'], user=self.getCurrentUser(), level=AccessType.READ)
+        allImages = [item for item in folderModel.childItems(folder) if 'largeImage' in item]
```

Comment:
This should probably check `if item.get('largeImage', {}).get('fileId')` or something equivalent.  An item can have a `largeImage` entry if it is pending being turned into a large image (for instance, being processed via vips), but will only have a `fileId` within the `largeImage` record if it actually has an image.